apisix/plugins/recaptcha.lua

```diff
@@ -0,0 +1,102 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local core = require("apisix.core")
+local http = require("resty.http")
+
+local schema = {
+    type = "object",
+    properties = {
+        secret_key = { type = "string" },
+        parameter_source = { type = "string", default = "header", enum = { "header", "query" } },
+        parameter_name = { type = "string", default = "captcha" },
+        ssl_verify = { type = "boolean", default = true },
+        response = {
+            type = "object",
+            properties = {
+                content_type = { type = "string", default = "application/json; charset=utf-8" },
+                status_code = { type = "number", default = 400 },
+                body = { type = "string", default = '{"message": "invalid captcha"}' }
+            }
+        },
+    },
+    required = { "secret_key" },
+}
+
+local recaptcha_url = "https://www.recaptcha.net"
+
+local _M = {
+    version = 0.1,
+    priority = 700,
+    name = "recaptcha",
+    schema = schema,
+}
+
+function _M.check_schema(conf, schema_type)
+    return core.schema.check(schema, conf)
```

Comment:
`schema_type` is unuesd?

Reply:
Please check this out. https://apisix.apache.org/docs/apisix/plugin-develop

At the same time, we need to implement the check_schema(conf) method to complete the specification verification.
```
function _M.check_schema(conf, schema_type)
    return core.schema.check(schema, conf)
end
```
Note: the project has provided the public method "core.schema.check", which can be used directly to complete JSON verification.